### PR TITLE
e2e: improve Watch test coverage

### PR DESCRIPTION
**Pull request by hexfusion**
ref: https://github.com/coreos/etcd/issues/9149

### CODE REVIEW

File: e2e/ctl_v3_lease_test.go [hunk 1]
```diff
@@ -138,21 +217,28 @@ func leaseTestKeepAliveOnce(cx ctlCtx) {
 	}
 }
 
-func leaseTestRevoke(cx ctlCtx) {
+func leaseTestRevoked(cx ctlCtx) {
```
Comment (gyuho) [inline e2e/ctl_v3_lease_test.go]: Why we have `leaseTestRevoked`? Just keep `leaseTestRevoke`?

File: e2e/ctl_v3_lease_test.go [hunk 2]
```diff
@@ -55,37 +126,45 @@ func leaseTestGrantTimeToLive(cx ctlCtx) {
 	}
 }
 
-func leaseTestGrantLeasesList(cx ctlCtx) {
+func leaseTestGrantLeaseListed(cx ctlCtx) {
```
Comment (gyuho) [inline e2e/ctl_v3_lease_test.go]: Why we have `leaseTestGrantLeaseListed`? Just keep `leaseTestGrantLeasesList`?

File: e2e/ctl_v3_auth_test.go [hunk 3]
```diff
@@ -952,7 +972,7 @@ func authTestSnapshot(cx ctlCtx) {
 	// ordinary user cannot save a snapshot
 	cx.user, cx.pass = "test-user", "pass"
 	if err := ctlV3SnapshotSave(cx, fpath); err == nil {
-		cx.t.Fatal("ordinary user should not be able to save a snapshot")
+		cx.t.Fatalf("ordinary user should not be able to save a snapshot")
```
Comment (gyuho) [inline e2e/ctl_v3_auth_test.go]: Just keep `Fatal`?

File: e2e/ctl_v3_lease_test.go (hunk 2 above)
Comment (hexfusion) [reply e2e/ctl_v3_lease_test.go]: `leaseTestGrantLeasesList` now returns error. I added leaseTestGrantLeaseListed for the local test to handle err Fatal?

File: e2e/ctl_v3_lease_test.go (hunk 1 above)
Comment (hexfusion) [reply e2e/ctl_v3_lease_test.go]: the same reasoning as `leaseTestGrantLeaseListed` will resolve if it does not make sense.

File: e2e/ctl_v3_auth_test.go (hunk 3 above)
Comment (hexfusion) [reply e2e/ctl_v3_auth_test.go]: agreed

File: e2e/ctl_v3_lease_test.go (hunk 1 above)
Comment (gyuho) [reply e2e/ctl_v3_lease_test.go]: I see. How about changing `leaseTestTimeToLiveExpire` to just error out, without returning error, and revert other changes?

Comment (hexfusion) [reply e2e/ctl_v3_lease_test.go]: I was only worried about handling the proper error content when used by other tests but this is probably getting ahead of myself as the nested nature will get kinda ugly. Will resolve.

Comment (hexfusion) [reply e2e/ctl_v3_lease_test.go]: If I don't return error in `leaseTestTimeToLiveExpire` how can I handle fatal below? I see this a lot in the tests where functions are reused, just making sure I understand your intent.

https://github.com/coreos/etcd/blob/22127895d818b161e7e96343d6b2745d9e23d879/e2e/ctl_v3_auth_test.go#L752-L764

Comment (gyuho) [reply e2e/ctl_v3_lease_test.go]: > If I don't return error in leaseTestTimeToLiveExpire how can I handle fatal below?

I don't understand.

The reason why we don't return an error, is that we want to know which line it's fatal-ing at.
If we return errors, the errors should be descriptive, as this PR does.

Either way works. Just want to make it consistent.

For example, `leaseTestRevoke` in this PR is mixing error returning and `cx.t.Errorf`.

Comment (hexfusion) [reply e2e/ctl_v3_lease_test.go]: understood, thanks

**Comment by hexfusion** [comment]
so I think this is more sane the cumulative errors should be like this `"parent_test: child_test error (%v)", err`. If we are on the same page I will make sure to enforce this moving forward.

**Comment by gyuho** [comment]
@hexfusion Sounds good. Thanks.

**Comment by hexfusion** [comment]
@gyuho so changes pushed meet your goal?

**Comment by hexfusion** [comment]
thanks!

**Pull request merged**